controller/workitem_blackbox_test.go

```diff
@@ -1210,6 +1210,38 @@ func (s *WorkItem2Suite) TestWI2ListByWorkitemtypeFilter() {
 	}
 }
 
+func (s *WorkItem2Suite) TestWI2ListByWorkitemstateFilter() {
+	// given
+	c := minimumRequiredCreatePayload()
+	c.Data.Attributes[workitem.SystemTitle] = "Title"
+	c.Data.Attributes[workitem.SystemState] = workitem.SystemStateNew
+	c.Data.Relationships = &app.WorkItemRelationships{
+		BaseType: &app.RelationBaseType{
+			Data: &app.BaseTypeData{
+				Type: "workitemtypes",
+				ID:   workitem.SystemBug,
+			},
+		},
+	}
+	// when
+	_, expected := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.wi2Ctrl, &c)
+	// then
+	assert.NotNil(s.T(), expected.Data)
+	require.NotNil(s.T(), expected.Data.ID)
+	require.NotNil(s.T(), expected.Data.Type)
+	require.NotNil(s.T(), expected.Data.Attributes)
+	wisNew := workitem.SystemStateNew
+	_, actual := test.ListWorkitemOK(s.T(), s.svc.Context, s.svc, s.wi2Ctrl, nil, nil, nil, nil, &wisNew, nil, nil, nil)
+
+	require.NotNil(s.T(), actual)
+	require.True(s.T(), len(actual.Data) > 1)
+	assert.Contains(s.T(), *actual.Links.First, fmt.Sprintf("filter[workitemstate]=%s", workitem.SystemStateNew))
+	for _, actualWI := range actual.Data {
+		assert.Equal(s.T(), expected.Data.Attributes[workitem.SystemState], actualWI.Attributes[workitem.SystemState])
+		require.NotNil(s.T(), actualWI.Attributes[workitem.SystemState])
+	}
```

Comment:
You could have 1 test case where you add x workitems with y workitems which have a specific state. 
And then you could assert whether there are exactly y number of workitems returned by that filter.